### PR TITLE
Adding extra testing for both net48 and net6.0 installers

### PR DESCRIPTION
**Pull request by sburmanoctopus**
[sc-58906]

# Background

We want to ensure our existing (net48) and new (net6.0) installers work with some automated tests.

# Results

Related to OctopusDeploy/Issues#8266

## Before
We have one existing automated test for our installers. It does actually cover both installers already. But all it tested was that the install location directory excluded a certain permission.

## After
We now also test that `Tentacle.exe` was installed, and that it can be run (just a simple command, `help`).

**Locally**
image

**TeamCity**
image

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: build/Build.Tests.cs [hunk 1]
```diff
@@ -194,10 +129,134 @@ bool DoesSidHaveRightsToDirectory(string directory, WellKnownSidType sid, params
...
+                    TestWindowsInstaller(installer);
                 }
             });
         });
+
+    string GetTestName(AbsolutePath installerPath) => Path.GetFileName(installerPath).Replace(".msi", "");
+        
+    void TestWindowsInstaller(AbsolutePath installerPath)
+    {
+        Log.Information($"\n--------------------------------------\nTesting Installer {GetTestName(installerPath)}\n--------------------------------------");
```
Comment (sburmanoctopus) [inline build/Build.Tests.cs:141]: Having this loud and proud was helpful during testing, so I left it. But am happy to dumb it down.

Comment (acodrington) [reply build/Build.Tests.cs:141]: I like the loud and proud :)

**Review by acodrington — APPROVED** [review]
Looks good. General refactor into nicely named methods helps readability too, nice job!
